portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,25 +34,50 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> comments = new ArrayList<>();
+  private static final String COMMENT_PARAM = "Comment";
+  private static final String TEXT_PROP = "text";
+  private static final String TIMESTAMP_PROP = "timestamp";
+  private static final String COMMENT_BOX_PARAM = "comment-box";
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = getUserComment(request);
-    comments.add(comment);
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity(COMMENT_PARAM);
+    commentEntity.setProperty(TEXT_PROP, comment);
+    commentEntity.setProperty(TIMESTAMP_PROP, timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+    
     response.sendRedirect("/index.html");
   }
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(COMMENT_PARAM).addSort(TIMESTAMP_PROP, SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
```

Comment:
This seems a bit sus because you're grabbing Datastore's internal identifier for the Entity. This seems okay if you just need a unique ID for each comment but since you're not using it yet I'm unsure whether the intended use makes sense or not.